SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,857 @@
+                        modOptionEventFound = true;
+                        parameters[i] = e;
+                    }
+
+                    if (senderFound && eventArgsFound && modOptionEventFound)
+                        break;
+                }
+
+                memberInfoMetadata.InvokeMethod(Config, parameters);
+            }
+        }
+
+        /// <summary>
+        /// Generates tooltips for each <see cref="ModOption"/> with a specified <see cref="TooltipAttribute"/>, before
+        /// invoking any relevant method(s) specified with <see cref="OnGameObjectCreatedAttribute"/>(s) and passes
+        /// parameters when a <see cref="GameObject"/> is created in the options menu.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void ConfigModOptions_GameObjectCreated(object sender, GameObjectCreatedEventArgs e)
+        {
+            if (modOptionsMetadata.TryGetValue(e.Id, out var modOptionMetadata))
+            {
+                // Create a tooltip if there is a TooltipAttribute specified
+                if (modOptionMetadata.ModOptionAttribute.Tooltip is string tooltip)
+                {
+                    e.GameObject.GetComponentInChildren<Text>().gameObject.AddComponent<ModOptionTooltip>().Tooltip = tooltip;
+                }
+
+                if (modOptionMetadata.OnGameObjectCreatedMetadata == null)
+                    return; // Skip attempting to invoke events if there are no OnGameObjectCreatedAttributes set for the member.
+
+                foreach (var onGameObjectCreatedMetadata in modOptionMetadata.OnGameObjectCreatedMetadata)
+                {
+                    InvokeEvent(onGameObjectCreatedMetadata, sender, e);
+                }
+            }
+        }
+        #endregion
+
+        #region Build ModOptions Menu
+        /// <summary>
+        /// Adds options to the menu based on the <see cref="modOptionsMetadata"/> dictionary, generated by the constructor.
+        /// </summary>
+        public override void BuildModOptions()
+        {
+            // Conditionally load the config
+            if (menuAttribute.LoadOn.HasFlag(MenuAttribute.LoadEvents.MenuOpened))
+                Config.Load();
+
+            foreach (var entry in modOptionsMetadata
+                .OrderBy(x => x.Value.ModOptionAttribute.Order)
+                .ThenBy(x => x.Value.MemberInfoMetadata.Name))
+            {
+                var id = entry.Key;
+                var modOptionMetadata = entry.Value;
+
+                Logger.Debug($"[ConfigModOptions] {modOptionMetadata.MemberInfoMetadata.Name}: " +
+                    $"{modOptionMetadata.ModOptionAttribute.GetType()}");
+                Logger.Debug($"[ConfigModOptions] Label: {modOptionMetadata.ModOptionAttribute.Label}");
+
+                var label = modOptionMetadata.ModOptionAttribute.Label;
+
+                if (modOptionMetadata.ModOptionAttribute is ButtonAttribute)
```

Comment:
This can be converted to a `switch` statement
```CS
switch (modOptionMetadata.ModOptionAttribute)
{
    case ButtonAttribute _:
        BuildModButtonOption(id, label);
        break;
    case ChoiceAttribute choiceAttribute:
        BuildModChoiceOption(id, label, modOptionMetadata.MemberInfoMetadata, choiceAttribute);
        break;
    case KeybindAttribute _:
        BuildModKeybindOption(id, label, modOptionMetadata.MemberInfoMetadata);
        break;
    case SliderAttribute sliderAttribute:
        BuildModSliderOption(id, label, modOptionMetadata.MemberInfoMetadata, sliderAttribute);
        break;
    case ToggleAttribute _:
        BuildModToggleOption(id, label, modOptionMetadata.MemberInfoMetadata);
        break;
}
```

Reply:
I prefer it too, I just hate the discard, but I guess its the same in the end. I'll convert.

Reply:
bbb3a18